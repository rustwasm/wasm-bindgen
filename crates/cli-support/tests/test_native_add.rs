const WASM: &[&str] = &[
    std::env!("CARGO_MANIFEST_DIR"),
    "/../../examples/wasm-in-wasm-imports/src/native_add.wasm",
];

mod common;

test_output_snapshot!(test_bundler_js, false, |b: &mut Bindgen| {
    b.bundler(true).unwrap();
});
test_output_empty!(test_bundler_start, true, |b: &mut Bindgen| {
    b.bundler(true).unwrap();
});
test_output_snapshot!(test_deno_js, false, |b: &mut Bindgen| {
    b.deno(true).unwrap();
});
test_output_empty!(test_deno_start, true, |b: &mut Bindgen| {
    b.deno(true).unwrap();
});
test_output_snapshot!(test_nodejs_js, false, |b: &mut Bindgen| {
    b.nodejs(true).unwrap();
});
test_output_empty!(test_nodejs_start, true, |b: &mut Bindgen| {
    b.nodejs(true).unwrap();
});
test_output_snapshot!(test_nodejs_module_js, false, |b: &mut Bindgen| {
    b.nodejs_module(true).unwrap();
});
test_output_snapshot!(test_nodejs_module_start, true, |b: &mut Bindgen| {
    b.nodejs_module(true).unwrap();
});
test_output_snapshot!(test_web_js, false, |b: &mut Bindgen| {
    b.web(true).unwrap();
});
test_output_empty!(test_web_start, true, |b: &mut Bindgen| {
    b.web(true).unwrap();
});
test_output_error!(test_nomodule_js, |b: &mut Bindgen| {
    // This test is expected to fail because the `no_modules` option cannot import
    b.no_modules(true).unwrap();
});
test_output_error!(test_nomodule_start, |b: &mut Bindgen| {
    // This test is expected to fail because the `no_modules` option cannot import
    b.no_modules(true).unwrap();
});
