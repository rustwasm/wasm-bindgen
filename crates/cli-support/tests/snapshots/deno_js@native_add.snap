---
source: crates/cli-support/tests/test_native_add.rs
expression: output.js()
---
import * as import0 from 'env'

const imports = {
  __wbindgen_placeholder__: {
	},
  'env': import0,
};

const wasm_url = new URL('native_add_bg.wasm', import.meta.url);
            let wasmCode = '';
            switch (wasm_url.protocol) {
                case 'file:':
                    wasmCode = await Deno.readFile(wasm_url);
                    break
                case 'https:':
                case 'http:':
                    wasmCode = await (await fetch(wasm_url)).arrayBuffer();
                    break
                default:
                    throw new Error(`Unsupported protocol: ${wasm_url.protocol}`);
            }

            const wasmInstance = (await WebAssembly.instantiate(wasmCode, imports)).instance;
            const wasm = wasmInstance.exports;
            export const __wasm = wasm;
